test/fuzz/transformation_swap_functions_test.cpp

```diff
@@ -0,0 +1,438 @@
+         %70 = OpLabel
+               OpReturn
+               OpFunctionEnd
+         
+         ; Function0
+         ; adjust type of the function in-place
+         %12 = OpFunction %8 None %9
+         %10 = OpFunctionParameter %7
+         %11 = OpFunctionParameter %7
+         %13 = OpLabel
+         %30 = OpLoad %6 %10
+         %32 = OpFDiv %6 %30 %31
+         %35 = OpLoad %6 %11
+         %36 = OpFDiv %6 %35 %31
+         %37 = OpFSub %6 %34 %36
+         %38 = OpCompositeConstruct %8 %32 %33 %37 %34
+               OpReturnValue %38
+               OpFunctionEnd
+         
+         ; Function1
+         %22 = OpFunction %8 None %18
+         %19 = OpFunctionParameter %7
+         %20 = OpFunctionParameter %15
+         %21 = OpFunctionParameter %17
+         %23 = OpLabel
+         %53 = OpVariable %7 Function
+         %54 = OpVariable %7 Function
+         %41 = OpLoad %6 %19
+         %44 = OpAccessChain %7 %21 %43
+         %45 = OpLoad %6 %44
+         %46 = OpFAdd %6 %41 %45
+         %47 = OpLoad %14 %20
+         %48 = OpConvertSToF %6 %47
+         %50 = OpAccessChain %7 %21 %49
+         %51 = OpLoad %6 %50
+         %52 = OpFAdd %6 %48 %51
+               OpStore %53 %46
+               OpStore %54 %52
+         %55 = OpFunctionCall %8 %12 %53 %54
+               OpReturnValue %55
+               OpFunctionEnd
+         
+         ; Function2
+         %28 = OpFunction %24 None %25
+         %26 = OpFunctionParameter %15
+         %27 = OpFunctionParameter %7
+         %29 = OpLabel
+         %58 = OpLoad %14 %26
+         %59 = OpConvertSToF %6 %58
+         %60 = OpLoad %6 %27
+         %61 = OpFOrdLessThan %24 %59 %60
+               OpReturnValue %61
+               OpFunctionEnd
+        
+  )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Try random permutations
+
+  ASSERT_TRUE(TransformationSwapFunctions({68, 87}).IsApplicable(
+      context.get(), transformation_context));
+
+  ASSERT_TRUE(TransformationSwapFunctions({68, 82}).IsApplicable(
+      context.get(), transformation_context));
+
+  ASSERT_TRUE(TransformationSwapFunctions({96, 104}).IsApplicable(
+      context.get(), transformation_context));
+
+  // Cannot swap a function with itself
+  ASSERT_FALSE(TransformationSwapFunctions({96, 96}).IsApplicable(
+      context.get(), transformation_context));
+  
+  // Function with id 10000 doesn't exist
+  ASSERT_FALSE(TransformationSwapFunctions({96, 1000}).IsApplicable(
+      context.get(), transformation_context));
+
+  std::string after_transformation = R"(
```

Comment:
Before we test that the module has changed, we must `Apply` some transformations first. (Note that `IsApplicable` on its own doesn't do anything). You can check other test files to see how it's done (e.g. `transformation_permute_function_parameters_test.cpp`).